File: code/Java/interpreter/command/OutputCommand.java
package interpreter.command;

import interpreter.expr.IntExpr;

public class OutputCommand extends Command {

    private IntExpr expr;

    public OutputCommand(int line, IntExpr expr) {
        super(line);
        this.expr = expr;
    }

    public void execute() {
        System.out.println(expr.expr());
    }

}

File: code/C++/interpreter/command/OutputCommand.cpp
#include <iostream>
#include "OutputCommand.h"
#include "../expr/IntExpr.h"

OutputCommand::OutputCommand(int line, IntExpr* expr)
	: Command(line), m_expr(expr) {
}

OutputCommand::~OutputCommand() {
	delete m_expr;
}

void OutputCommand::execute() {
	int v = m_expr->expr();
	std::cout << v << std::endl;
}
